### ISSUE TITLE
ERROR - this language feature is only supported for ECMASCRIPT6 mode or better: block-scoped function declaration

### ISSUE THREAD

**Issue by Vancer** [google/blockly-games#74]
Hello team!

Before I proceed, I took a look on the issue #52  which is the same.
I've tried to check the `workspace_svg.js` file but there are no constants, assuming that you have already fixed it too [Issue was back in 2017].

Any way, this is the error that I'm getting when trying to compile using the command `python build-app.py maze en` . Note that I am on MacOS Sierra 10.12.6.

Maze - en - uncompressed:
closure-library/closure/bin/build/closurebuilder.py: Scanning paths...
closure-library/closure/bin/build/closurebuilder.py: 2424 sources scanned.
closure-library/closure/bin/build/closurebuilder.py: Building dependency tree..

Maze - en - compressed:
appengine/third-party/SoundJS/soundjs-NEXT.js:310: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:311: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:312: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:314: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:315: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:316: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:317: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:2400: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:4023: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:6819: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:7046: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:7106: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:7441: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs-NEXT.js:7459: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:310: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:311: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:312: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:314: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:315: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:316: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:317: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:2400: WARNING - Bad type annotation. type annotation incompatible with other annotations. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:4023: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:6819: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:7046: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:7106: WARNING - Bad type annotation. expected closing } See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:7441: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/SoundJS/soundjs.js:7459: WARNING - Bad type annotation. type not recognized due to syntax error. See https://github.com/google/closure-compiler/wiki/Bad-Type-Annotation for more information.

appengine/third-party/midi-js/examples/inc/timer.js:187: WARNING - String continuations are not recommended. See https://google.github.io/styleguide/javascriptguide.xml?showone=Multiline_string_literals#Multiline_string_literals

appengine/third-party/midi-js/generator/nodejs/packageWAVs.js:87: WARNING - String continuations are not recommended. See https://google.github.io/styleguide/javascriptguide.xml?showone=Multiline_string_literals#Multiline_string_literals

appengine/third-party/ace/worker-xml.js:1: ERROR - this language feature is only supported for ECMASCRIPT6 mode or better: block-scoped function declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT or higher to enable ES6 features.

1 error(s), 30 warning(s)

**Comment by NeilFraser** [comment]
Yes, the error is in a third-party dependency:
appengine/third-party/ace/worker-xml.js

We submitted a PR to them last week, which they accepted:
https://github.com/ajaxorg/ace/pull/3563

Just waiting on them to repackage this file:
https://github.com/ajaxorg/ace-builds/blob/master/src-min-noconflict/mode-xml.js

**Comment by NeilFraser** [comment]
Ace repackaged the file, but the changes were not part of the new build.  I've asked why:
https://github.com/ajaxorg/ace-builds/issues/123

**Comment by Vancer** [comment]
@NeilFraser  Thanks! I will wait until it's part of the new build! Thanks.

**Comment by alianglan** [comment]
hello team，it is still can't build

**Comment by alianglan** [comment]
Hello team, is this issue resovled ?

**Comment by NeilFraser** [comment]
I pinged ACE again, and they just fixed it.

Blockly Games appears to compile properly now, with the exception of the Music game (this is due to an upstream issue in Blockly, tracked in #77).
